remoteappmanager/rest/rest_handler.py

```diff
@@ -25,6 +26,44 @@ def get_resource_handler_or_404(self, collection_name):
         except KeyError:
             raise web.HTTPError(httpstatus.NOT_FOUND)
 
+    def write_error(self, status_code, **kwargs):
+        """Provides appropriate payload to the response in case of error.
+        """
+        exc_info = kwargs.get("exc_info")
+
+        if exc_info is None:
+            self.clear_header('Content-Type')
+            self.finish()
+
+        exc = exc_info[1]
+
+        if isinstance(exc, PayloadedHTTPError) and exc.payload is not None:
```

Comment:
> My vision is that this method only handles http-level exception types, with the conversion of exception type done at the individual http method handlers.

We can't prevent non-http-level exception from arriving at write_error.  write_error is intended to be able to handle BaseException in general.

>  This allows us to take appropriate action and, for example log contextually to the method handler. Converting here we lose this specificity.

I don't think converting a RestException to PayloadedHTTPError here stop you from logging contextually.  If you must log, you can do so in Rest*Handler
```
try:
   yield method()
except Exception:
   log.exception
   raise
```